internal/controllers/device_data_controller.go

```diff
@@ -407,6 +408,71 @@ func (udc *UserDevicesController) GetUserDeviceErrorCodeQueries(c *fiber.Ctx) er
 	return c.JSON(GetUserDeviceErrorCodeQueriesResponse{Queries: queries})
 }
 
+// ClearUserDeviceErrorCodeQuery godoc
+// @Description Clear user device error codes
+// @Tags        user-devices
+// @Success     200 {object} controllers.QueryDeviceErrorCodesResponse
+// @Security    BearerAuth
+// @Router      /user/devices/{userDeviceID}/error-codes/clear [post]
+func (udc *UserDevicesController) ClearUserDeviceErrorCodeQuery(c *fiber.Ctx) error {
+	userID := helpers.GetUserID(c)
+	udi := c.Params("userDeviceID")
+
+	logger := helpers.GetLogger(c, udc.log)
+
+	userDeviceExists, err := models.UserDevices(
+		models.UserDeviceWhere.UserID.EQ(userID),
+		models.UserDeviceWhere.ID.EQ(udi),
+	).Exists(c.Context(), udc.DBS().Reader)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return fiber.NewError(fiber.StatusNotFound, err.Error())
+		}
+
+		logger.Err(err).Msg("error occurred when fetching error codes for device")
+		return fiber.NewError(fiber.StatusInternalServerError, "error occurred updating device error queries")
+	}
+
+	if !userDeviceExists {
+		return fiber.NewError(fiber.StatusUnauthorized, "error, user does not own device")
+	}
+
+	/*
+		// TODO - I'm not so sure clearing the last item without any input from the caller is such a good idea based on the UI on figma
+
+		req := &ClearUserDeviceErrorCodeQueryReq{}
+		   	if err := c.BodyParser(req); err != nil {
+		   		return fiber.NewError(fiber.StatusBadRequest, "Couldn't parse request.")
+		   	}
+	*/
+
+	errCodeQuery, err := models.ErrorCodeQueries(
+		models.ErrorCodeQueryWhere.UserDeviceID.EQ(udi),
+		models.ErrorCodeQueryWhere.ClearedAt.IsNull(),
```

Comment:
I would drop this. If the most recent query has been cleared, then I don’t think this endpoint has anything to do..